File: src/dotnet/PloyBus.Tests/Transport/Transactions/Messages/Handlers/Error/TestBus.cs
namespace PolyBus.Transport.Transactions.Messages.Handlers.Error;

class TestBus(string name) : IPolyBus
{
    public IDictionary<string, object> Properties { get; } = new Dictionary<string, object>();
    public ITransport Transport { get; } = new TestTransport();
    public IList<IncomingHandler> IncomingPipeline { get; } = [];
    public IList<OutgoingHandler> OutgoingPipeline { get; } = [];
    public Messages Messages { get; } = new();
    public Task<IncomingTransaction> CreateIncomingTransaction(IncomingMessage message) =>
        Task.FromResult(new IncomingTransaction(this, message));
    public Task<OutgoingTransaction> CreateOutgoingTransaction() =>
        Task.FromResult(new OutgoingTransaction(this));
    public string Name { get; } = name;
    public Task Send(Transaction transaction) => Task.CompletedTask;
    public Task Start() => Task.CompletedTask;
    public Task Stop() => Task.CompletedTask;
}

File: src/kotlin/src/test/kotlin/polybus/transport/transactions/messages/handlers/error/TestBus.kt
package polybus.transport.transactions.messages.handlers.error

import polybus.IPolyBus
import polybus.transport.ITransport
import polybus.transport.transactions.IncomingTransaction
import polybus.transport.transactions.OutgoingTransaction
import polybus.transport.transactions.Transaction
import polybus.transport.transactions.messages.IncomingMessage
import polybus.transport.transactions.messages.Messages
import polybus.transport.transactions.messages.handlers.IncomingHandler
import polybus.transport.transactions.messages.handlers.OutgoingHandler

class TestBus(override val name: String) : IPolyBus {
    override val properties: MutableMap<String, Any> = mutableMapOf()
    override var transport: ITransport = TestTransport()
    override val incomingPipeline: MutableList<IncomingHandler> = mutableListOf()
    override val outgoingPipeline: MutableList<OutgoingHandler> = mutableListOf()
    override val messages: Messages = Messages()

    override suspend fun createIncomingTransaction(message: IncomingMessage): IncomingTransaction =
        IncomingTransaction(this, message)

    override suspend fun createOutgoingTransaction(): OutgoingTransaction = OutgoingTransaction(this)

    override suspend fun send(transaction: Transaction) {}
    override suspend fun start() {}
    override suspend fun stop() {}
}
